scipy/sparse/linalg/_isolve/iterative.py

```diff
@@ -670,14 +671,14 @@ def gmres(A, b, x0=None, tol=None, restart=None, maxiter=None, M=None,
 
     # Handle the deprecation frenzy
     if restrt and restart:
-        raise ValueError("Cannot specify both restart and restrt"
+        raise ValueError("Cannot specify both 'restart' and 'restrt'"
                          " keywords. Also 'rstrt' is deprecated."
-                         " and will be removed in SciPy 1.12.0. Use "
+                         " and will be removed in SciPy 1.14.0. Use "
                          "'restart' instad.")
-    if restrt is not None:
+    if restrt is not _NoValue:
         msg = ("'gmres' keyword argument 'restrt' is deprecated "
                "in favor of 'restart' and will be removed in SciPy"
-               " 1.12.0. Until then, if set, 'rstrt' will override 'restart'."
+               " 1.14.0. Until then, if set, 'rstrt' will override 'restart'."
```

Comment:
Version in the docstring also needs bumping

Reply:
One of these days, I'll start with coffee first before touching deprecations.